### PR TITLE
[hdf5] Update to 1.14.3 and fix compilation error under arm64-windows

### LINKED ISSUE #24898: [hdf5] Build error for Windows Arm64 platforms
**Host Environment**
- OS: Windows
- Compiler: VS 2022

**To Reproduce**
`.\vcpkg.exe install hdf5 --triplet=arm64-windows`

**Failure logs**

From CMake error log

```
CMake Error at config/cmake/ConfigureChecks.cmake:334 (file):
  file failed to open for reading (No such file or directory):

    C:/Users/niysai01/Workspace/vcpkg/buildtrees/hdf5/arm64-windows-dbg/pac_Cconftest.out
Call Stack (most recent call first):
  CMakeLists.txt:464 (include)
```

** Additional Context **

Static builds seem to be working

`.\vcpkg.exe install hdf5 --triplet=arm64-windows-static`

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/24898 
Update the version of `hdf5` to 1.14.3.
Update the version of `openturns` to 1.22.
Remove warning in `netgen`.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x64-windows
arm64-windows
```
Compile test pass with following triplets:
```
arm64-android
x64-windows
arm64-windows
```

### CODE REVIEW

File: ports/hdf5/portfile.cmake [hunk 1]
```diff
@@ -5,12 +5,13 @@ vcpkg_from_github(
...
     REPO  HDFGroup/hdf5
     REF "${hdf5_ref}"
-    SHA512 9b44993bcdc1493a22da61c77a1bd962c0088ff8e7fb75c00568617386cfc296a73bbdae79c05847109bf1984e95660bbe459f8a96950f6cf71002800eed23f8
+    SHA512 4c74d2a1499a5f8bfcc1bf32b5901e413ccf46714a5ccdb4e190f51479c96a0d05ce47aae700945a45dc69f3bb0f3891c02bb522abd29cf16e8cb3a2fea8013c
     HEAD_REF develop
     PATCHES
         hdf5_config.patch
         szip.patch
         pkgconfig-requires.patch
+        fix_arm_build.patch #https://github.com/HDFGroup/hdf5/pull/3683 https://github.com/HDFGroup/hdf5/pull/4160
```
[the remark this answers is not in the document]
Comment (jimwang118) [reply ports/hdf5/portfile.cmake:14]: Directly downloading the PR patch will cause patch conflicts and errors will occur during compilation.

File: ports/netgen/fix_warning.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
++#include <iostream>
+ 
+ #include "ngcore_api.hpp"       // for NGCORE_API and CPU arch macros
+ 
+@@ -74,7 +75,7 @@ namespace ngcore
+ #elif defined(__EMSCRIPTEN__)
+     return std::chrono::high_resolution_clock::now().time_since_epoch().count();
+ #else
+-#warning "Unsupported CPU architecture"
++    std::cout << "Unsupported CPU architecture" << std::endl;
```
Comment (dg0yt) [inline ports/netgen/fix_warning.patch:18]: What is the aim of this patch? It changes observable behavior in a way that may break things, but doesn't fix anything AFAICS.

Comment (jimwang118) [reply ports/netgen/fix_warning.patch:18]: This #warning is not recognized under arm64-windows, so an error will be included during compilation.
`D:\b\netgen\src\v6.2.2401-9b711d5bba.clean\libsrc\core\utils.hpp(77): fatal error C1021: invalid preprocessor command 'warning'`

Comment (dg0yt) [reply ports/netgen/fix_warning.patch:18]: Yeah, and then it is okay to remove or replace the build time `#warning`.
But it is not okay to add expensive runtime `cout` stuff to a cheap inline `GetTimeCounter` function when upstream intentionally makes it an ultra-cheap no-op. How many times is called per second? Where does it print the output to?

Comment (jimwang118) [reply ports/netgen/fix_warning.patch:18]: Yes, he creates a waste of time. So should I just delete it?

Comment (dg0yt) [reply ports/netgen/fix_warning.patch:18]: Yes.

**Comment by LilyWangLL** [comment]
PR #38911 has updated `hdf5` to 1.14.4.3, this PR could be closed.

**Pull request closed without merging**